src/Polytopes/iterators.jl

```diff
@@ -0,0 +1,227 @@
+end
+
+struct Cone #a real polymake polyhedron
+    pm_cone::Polymake.BigObject
+end
+
+struct PointVector{U} <: AbstractVector{U}
+    p::Polymake.Vector
+    PointVector{U}(p) where U = new(Polymake.Vector{Polymake.to_cxx_type(U)}(p))
+end
+
+Base.IndexStyle(::Type{<:PointVector}) = IndexLinear()
+
+Base.getindex(po::PointVector, i::Base.Integer) = po.p[i]
+
+function Base.setindex!(po::PointVector{U}, val, i::Base.Integer) where U
+    @boundscheck checkbounds(po.p, i)
+    po.p[i] = val
+    return val
+end
+
+Base.firstindex(::PointVector) = 1
+Base.lastindex(iter::PointVector) = length(iter)
+Base.size(po::PointVector) = size(po.p)
+
+PointVector(x...) = PointVector{Polymake.Rational}(x...)
+
+PointVector{U}(n::Base.Integer) where U = PointVector{U}(Polymake.Vector{U}(undef, Polymake.Integer(n)))
+
+function Base.similar(X::PointVector, ::Type{S}, dims::Dims{1}) where S <: Union{Polymake.Rational, Polymake.Integer, Int64, Float64}
+    return PointVector{S}(dims...)
+end
+
+Base.BroadcastStyle(::Type{<:PointVector}) = Broadcast.ArrayStyle{PointVector}()
+
+function Base.similar(bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{PointVector}}, ::Type{ElType}) where ElType
+    return PointVector{Polymake.promote_to_pm_type(Vector, ElType)}(axes(bc)...)
+end
+
+struct RayVector{U} <: AbstractVector{U}
+    p::Polymake.Vector
+    RayVector{U}(p) where U = new(Polymake.Vector{Polymake.to_cxx_type(U)}(p))
+end
+
+Base.IndexStyle(::Type{<:RayVector}) = IndexLinear()
+
+Base.getindex(po::RayVector, i::Base.Integer) = po.p[i]
+
+function Base.setindex!(po::RayVector{U}, val, i::Base.Integer) where U
+    @boundscheck checkbounds(po.p, i)
+    po.p[i] = val
+    return val
+end
+
+Base.firstindex(::RayVector) = 1
+Base.lastindex(iter::RayVector) = length(iter)
+Base.size(po::RayVector) = size(po.p)
+
+RayVector(x...) = RayVector{Polymake.Rational}(x...)
+
+RayVector{U}(n::Base.Integer) where U = RayVector{U}(Polymake.Vector{U}(undef, Polymake.Integer(n)))
+
+function Base.similar(X::RayVector, ::Type{S}, dims::Dims{1}) where S <: Union{Polymake.Rational, Polymake.Integer, Int64, Float64}
+    return RayVector{S}(dims...)
+end
+
+Base.BroadcastStyle(::Type{<:RayVector}) = Broadcast.ArrayStyle{RayVector}()
+
+function Base.similar(bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{RayVector}}, ::Type{ElType}) where ElType
+    return RayVector{Polymake.promote_to_pm_type(Vector, ElType)}(axes(bc)...)
+end
+
+@doc Markdown.doc"""
+    Halfspace(a, b)
+
+One halfspace `H(a,b)` is given by a vector `a` and a value `b` such that
+$$H(a,b) = \{ x | ax ≤ b \}.$$
+"""
+struct Halfspace
+    a::Polymake.Matrix{Polymake.Rational}
+    b::Polymake.Rational
+end
+
+Halfspace(a::AbstractVector, b) = Halfspace(reshape(a, 1, :), b)
+
+Halfspace(a) = Halfspace(a, 0)
+
+# TODO: abstract notion of equality
+Base.:(==)(x::Halfspace, y::Halfspace) = x.a == y.a && x.b == y.b
+
+struct PolyhedronOrConeIterator{T} <: AbstractArray{T, 1}
```

Comment:
```suggestion
struct PolyhedronOrConeIterator{T} <: AbstractVector{T}
```
Please also look for any other users of `AbtractArray` or `Array` and replace them by `(Abstract){Vector,Matrix}` as appropriate.